examples/streamlit/src/run_app.py

```diff
@@ -0,0 +1,36 @@
+import os
+import subprocess
+import sys
+import webbrowser
+
+from src.config import EnvironmentalVariableNames as EnvVar, get_env
+
+
+def main():
+
+    # Getting path to python executable (full path of deployed python on Windows)
+    executable = sys.executable
+
+    # Open browser tab. May temporarily display error until streamlit server is started.
+    command = 'import time; import webbrowser; time.sleep(3); webbrowser.open("http://localhost:8501")'
+    subprocess.run(f"{executable} -c '{command}'", shell=True)
+
+    # Run streamlit server
+    path_to_main = os.path.join(
+        get_env(EnvVar.WORKING_DIR), "src", "app.py"
+    )
+    result = subprocess.run(
+        f"{executable} -m streamlit run {path_to_main} --server.headless=true --global.developmentMode=false",
```

Comment:
I'm a bit puzzled why you need `--server.headless=true` and calling `webbrowser.open()` yourself. Streamlit should [call the same thing itself](https://github.com/streamlit/streamlit/blob/4edc980e513cac8dac07aa68ad0024ec29063a5a/lib/streamlit/util.py#L88) - and it can do this when the server has started up, rather than hoping that 3 seconds wait will work. :slightly_smiling_face: 

What do you see if you don't pass this option?

Reply:
Actually I did not try without passing the `--server.heaedless=true` option. But I agree that it should not be necessesary. 

About the `webbrowser.open`, it is there because when I tried to run my app on Windows, the browser did not open automatically. I added the waiting of three second because I noticed that the first time it is run, streamlit takes a while to start and, if the browser tab is open immediately, it says that the `localhost:8501` is not reachable

Reply:
I think `--server.headless=true` tells it not to open the web browser itself, so if you remove that and the `webbrowser.open`, it *should* work. It would be good to try that.

If you do have to launch the browser yourself, a short wait makes sense, because there's probably no easy way to know when the server is ready. But that's the advantage of letting streamlit open the browser if possible.

Reply:
I agree with you that streamlit should start without any additional argument, but I did some trials and indeed the `--server.headless=true` argument seems to be necessary, otherwise the streamlit server does not start. I left it in the argument list passed to subprocess.

About the opening of the browser tab, I also verified that it is not enough to remove both the `webbrowser.open` and the `--server.headless=true`. Actually neither the version that is pushed here works, in which I call `webbrowser.open` inside another `subprocess.run`. I left two lines of code waiting for a brief time and then explicitly opening the tab after the streamlit server is launched. We may add a comment to notify the reason why they are there.

Reply:
Fair enough, thanks. Yes, let's add a comment explaining about this, and maybe inviting someone to figure out why `--server.headless=true` doesn't work, because I still think it *should*, but I don't have the time to dig in and figure out why it doesn't. :wink:

Reply:
Added comments. Maybe we can open an issue on streamlit github or on discuss.streamlit.io to notify this fact.

Reply:
Thanks! I'll let you do that, as I'm not familiar with Streamlit at all, but I'm happy to pitch in with details about Pynsist if needed. If you're not sure which site to start it on, I'd suggest the discussion one - I'd only go straight to the issue tracker if I'm pretty confident that there's something wrong in streamlit.

(I see they've got a message up saying the team will be away for the US thanksgiving holiday, though, so you might want to wait a few days. :-)